modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/OperationMessageProcessor.java

```diff
@@ -135,44 +134,26 @@ public Event process(Event event) throws MuleException {
       ComponentIdentifier operationIdentifier =
           new ComponentIdentifier.Builder().withName(operationModel.getName())
               .withNamespace(extensionModel.getName().toLowerCase()).build();
-      Map<String, Object> originalParameterMap = this.resolverSet.resolve(event).asMap();
-      PolicyProvider policyProvider = policyManager.lookupPolicyProvider();
-      Optional<Policy> policy =
-          policyProvider.findOperationPolicy(event.getContext().getId(), operationIdentifier);
-      if (policy.isPresent()) {
-        Optional<PolicyOperationParametersTransformer> policyOperationParametersTransformer =
+      Optional<OperationPolicy> policy = policyManager.findOperationPolicy(event.getContext().getId(), operationIdentifier);
```

Comment:
Seems to me that finding the policy on each process() could hurt performance. I don't know if this is possible, but would it make sense to fetch the policy on start(), and then derive a new execution of that policy on each process. What I mean is that instead of doing `findPolicy(eventId, operationIdentifier)`, you could findPolicy only once and then do `policy.newExecution(eventId)`. Another advantage of this would be that you no longer need to make sure that the execution state gets disposed. It can just be collected.